src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -16,18 +30,74 @@
      * and returns a FindCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DEPARTMENT, PREFIX_OFFICE,
+                        PREFIX_REMARK, PREFIX_TAG);
+
+        if (!areAnyPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_DEPARTMENT, PREFIX_OFFICE,
+                PREFIX_REMARK, PREFIX_TAG)) {
+
+            String trimmedArgs = args.trim();
+            if (trimmedArgs.isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            }
+
+            String[] nameKeywords = trimmedArgs.split("\\s+");
+
+            return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        }
+
+        String[] keywords;
+        List<Predicate<Person>> predicates = new ArrayList<>();
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            keywords = argMultimap.getValue(PREFIX_NAME).get().split("\\s+");
+            predicates.add(new NameContainsKeywordsPredicate(Arrays.asList(keywords)));
+        }
+
+        if (argMultimap.getValue(PREFIX_DEPARTMENT).isPresent()) {
+            keywords = argMultimap.getValue(PREFIX_DEPARTMENT).get().split("\\s+");
+            predicates.add(new DepartmentContainsKeywordsPredicate(Arrays.asList(keywords)));
+        }
+
+        if (argMultimap.getValue(PREFIX_OFFICE).isPresent()) {
+            keywords = argMultimap.getValue(PREFIX_OFFICE).get().split("\\s+");
+            predicates.add(new OfficeContainsKeywordsPredicate(Arrays.asList(keywords)));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        if (argMultimap.getValue(PREFIX_REMARK).isPresent()) {
+            keywords = argMultimap.getValue(PREFIX_REMARK).get().split("\\s+");
+            predicates.add(new RemarkContainsKeywordsPredicate(Arrays.asList(keywords)));
+        }
+
+        if (argMultimap.getValue(PREFIX_TAG).isPresent()) {
+            keywords = argMultimap.getValue(PREFIX_TAG).get().split("\\s+");
+            predicates.add(new TagContainsKeywordsPredicate(Arrays.asList(keywords)));
+        }
+
+        if (predicates.size() == 1) {
+            return new FindCommand(predicates.get(0));
+        }
+
+        Predicate<Person> composedPredicate = person -> true;
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        for (Predicate<Person> p: predicates) {
```

Comment:
Should we extract this out as a method? To follow the SLAP principle

Reply:
Ok! 👍🏻